pkg/investigations/investigation/investigation.go

```diff
@@ -20,16 +20,12 @@ type InvestigationResult struct {
 	ServiceLogSent     InvestigationStep
 }
 
-// Investigation serves as a parent class
-// This enables the structure of cmd/investigate.go
-type Investigation struct {
-	Run  func(resources *Resources) (InvestigationResult, error)
-	Name string
-}
-
-// NewInvestigation creates a new investigation
-func NewInvestigation(investigationFn func(resources *Resources) (InvestigationResult, error), name string) *Investigation {
-	return &Investigation{investigationFn, name}
+type Investigation interface {
+	Investigate(resources *Resources) (InvestigationResult, error)
```

Comment:
Do you think calling this function `Run` is "cleaner" than `Investigate`? This reduces repetitiveness e.g. if you have 
`investigation.Investigate` vs `investigation.Run`